src/rect.rs

```diff
@@ -103,6 +103,19 @@ impl<T, U> TypedRect<T, U> {
     }
 }
 
+impl<T, U> TypedRect<T, U>
+where
+    T: Copy + Zero
```

Comment:
I don't think `Copy` is needed here

Reply:
Ah, yes indeed.

Reply:
Err, actually, truns out Zero requires Copy.